src/cmd/dev.go

```diff
@@ -163,11 +163,16 @@ var extractCmd = &cobra.Command{
 			return err
 		}
 
-		// NOTE: I am obviously not chcking any of the outputs here, but this call is what sets
-		//       bndleClient.cfg.DeployOpts.Source which we need later...
-		_, _, _, err = bndlClient.PreDeployValidation()
-		if err != nil {
-			return err
+		if bundleCfg.IsTofu {
+			_, _, _, err = bndlClient.PreDeployValidationTF()
+			if err != nil {
+				return err
+			}
+		} else {
+			_, _, _, err = bndlClient.PreDeployValidation()
+			if err != nil {
+				return err
+			}
 		}
```

Comment:
nit: same style nit from other pr; this just reduces nested `if err{}` blocks

```suggestion
		if bundleCfg.IsTofu {
			_, _, _, err = bndlClient.PreDeployValidationTF()
		} else {
			_, _, _, err = bndlClient.PreDeployValidation()
		}
		if err != nil {
			return err
		}
```